src/Disciplina/DB/DSL/SimpleTxDB.hs

```diff
@@ -0,0 +1,155 @@
+module Disciplina.DB.DSL.SimpleTxDB () where
+
+import Universum
+
+import Codec.Serialise (Serialise(..))
+import Crypto.Error (CryptoFailable(..))
+
+import Disciplina.Crypto.Signing.Class (AbstractPK(..), AbstractSK(..))
+import Disciplina.DB.DSL.Types (QueryTx(..), QueryTxs(..), WHERE(..)
+                               ,TxIdEq(..), TxGrade(..), QueryObj(..)
+                               ,TxsFilterExpr(..), ObjHashEq(..))
+import Disciplina.Educator.Txs (PrivateTxId, PrivateTx(..), EducatorTxMsg(..)
+                               ,StudentTxMsg(..), PrivateTxPayload(..))
+import Disciplina.Crypto (hash, PublicKey, SecretKey)
+import Disciplina.Core (Address(..), CourseId(..), SubjectId, Grade(..)
+                       ,mkAddr, hasPathFromTo
+                       ,activityTypeGraphIndexed)
+import Disciplina.DB.DSL.Interpret (MonadSearchTxObj(..), RunQuery(..))
+import Data.List (union, intersect)
+
+import qualified Crypto.PubKey.Ed25519 as Ed25519
+import qualified Data.ByteString.Char8 as C
+
+
+-- | Simple transaction database
+newtype SimpleTxDB a = SimpleTxDB
+   { getSimpleTxDB :: Reader [PrivateTx] a
+   } deriving (Functor, Applicative, Monad, MonadReader [PrivateTx])
+
+instance MonadSearchTxObj SimpleTxDB where
+  runTxQuery = runSimpleTxQuery
+  runTxsQuery = runSimpleTxsQuery
+  runObjQuery = undefined -- TODO, implement this one, but how are Object defined?
+
+instance Serialise Address where
+  encode (Address h) = encode h
+  decode = Address <$> decode
+
+-- | TODO, where should we put these?
```

Comment:
Separate 'Serialise' modules, I think. Soon there will be my PR on DSCP-61, you'll see how I see it.